dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackerTrigramIndexingJob.java

```diff
@@ -0,0 +1,145 @@
+    {
+        TrackerTrigramIndexJobParameters parameters = (TrackerTrigramIndexJobParameters) jobConfiguration
+            .getJobParameters();
+
+        log.info( "Starting Trigram Indexing Job. Attributes Provided to Index: {}", parameters.getAttributes() );
+        progress.startingProcess( "Starting Trigram indexing process" );
+
+        if ( !CollectionUtils.isEmpty( parameters.getAttributes() ) )
+        {
+            log.debug( "Fetching all indexable attributes from db" );
+            Set<TrackedEntityAttribute> indexableAttributes = trackedEntityAttributeService
+                .getAllTrigramIndexableTrackedEntityAttributes();
+
+            log.debug( "Found total {} indexable attributes", indexableAttributes.size() );
+            indexableAttributes = indexableAttributes.stream()
+                .filter( itea -> parameters.getAttributes().contains( itea.getUid() ) ).collect( Collectors.toSet() );
+
+            log.debug( "Number of Attributes provided in jobparameters that are indexable: {}",
+                indexableAttributes.size() );
+
+            if ( !indexableAttributes.isEmpty() )
+            {
+                log.debug( "Creating {} trigram indexes", indexableAttributes.size() );
+                progress.startingStage( "Creating trigram indexes for attributes", indexableAttributes.size() );
+                progress.runStage( indexableAttributes.stream(), TrackedEntityAttribute::getName,
+                    tea -> trackedEntityAttributeService.createTrigramIndex( tea ),
+                    TrackerTrigramIndexingJob::computeTrigramIndexingSummary );
+                progress.completedStage( "Trigram indexes created" );
+                log.debug( "Created {} trigram indexes", indexableAttributes.size() );
+            }
+            else
+            {
+                log.warn( "No indexable attributes provided in jobParameters. Skipping trigram index creation step" );
+            }
+        }
+
+        if ( parameters.isRunVacuum() )
+        {
+            log.debug( "Running VACUUM" );
+            progress.startingStage( "Running VACUUM on tracker tables", 1 );
+            progress.runStage( () -> trackedEntityAttributeService.runVacuum() );
+            progress.completedStage( "VACUUM run completed" );
+            log.debug( "VACUUM completed" );
+        }
+
+        if ( parameters.isRunAnalyze() )
+        {
+            log.debug( "Running ANALYZE" );
+            progress.startingStage( "Running ANALYZE on tracker tables", 1 );
+            progress.runStage( () -> trackedEntityAttributeService.runAnalyze() );
+            progress.completedStage( "ANALYZE run completed" );
+            log.debug( "ANALYZE completed" );
+        }
+
+        progress.completedProcess( "Job completed" );
+        log.info( "Trigram Indexing Job completed" );
+    }
+
+    private static String computeTrigramIndexingSummary( int successful, int failed )
+    {
+        log.info(
```

Comment:
I would split logging and creation of a summary. Seem like two concerns to me.

Is it intentional that the summary returned does not contain the `successful` ones but only the `failed` whereas the logs do use both?

Maybe you can compute the summary for success/failure, then outside of here pass the summary to a log statement

Reply:
Modified this a bit.